File: android/app/src/main/java/com/bunty/clipsync/LocationHelper.kt
package com.bunty.clipsync

import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.json.JSONObject
import java.net.HttpURLConnection
import java.net.URL
import java.io.BufferedReader
import java.io.InputStreamReader

object LocationHelper {

    // --- IP Location Detection ---
    // Used to route user to the nearest Cloud Functions region (US vs IN)
    suspend fun detectCountryCode(): String? {
        return withContext(Dispatchers.IO) {
            try {
                // Using ip-api.com (free, non-commercial use)
                // Note: In production, consider a paid service or robust fallback
                val url = URL("http://ip-api.com/json/") 
                val connection = url.openConnection() as HttpURLConnection
                connection.requestMethod = "GET"
                connection.connectTimeout = 5000
                connection.readTimeout = 5000

                if (connection.responseCode == 200) {
                    val reader = BufferedReader(InputStreamReader(connection.inputStream))
                    val response = StringBuilder()
                    var line: String? 
                    while (reader.readLine().also { line = it } != null) {
                        response.append(line)
                    }
                    reader.close()
                    connection.disconnect()

                    val json = JSONObject(response.toString())
                    val countryCode = json.optString("countryCode", "")
                    Log.d("LocationHelper", "Detected Country: $countryCode")
                    countryCode
                } else {
                    Log.e("LocationHelper", "Failed to get location: ${connection.responseCode}")
                    null
                }
            } catch (e: Exception) {
                Log.e("LocationHelper", "Error detecting location", e)
                null
            }
        }
    }
}

File: mac/ClipSync/LocationHelper.swift
//
//  LocationHelper.swift
//  ClipSync
//

import Foundation

class LocationHelper {
    static let shared = LocationHelper()
    
    // --- IP Location Detection ---
    func detectRegion(completion: @escaping (String?) -> Void) {
        guard let url = URL(string: "https://ip-api.com/json/") else {
            completion(nil)
            return
        }
        
        let task = URLSession.shared.dataTask(with: url) { data, response, error in
            guard let data = data, error == nil else {
                print("❌ Location detection failed: \(error?.localizedDescription ?? "Unknown error")")
                completion(nil)
                return
            }
            
            do {
                if let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
                   let countryCode = json["countryCode"] as? String {
                    print("📍 Detected Country: \(countryCode)")
                    completion(countryCode)
                } else {
                    completion(nil)
                }
            } catch {
                print("❌ JSON parse error: \(error)")
                completion(nil)
            }
        }
        task.resume()
    }
}
